test/utils.jl

```diff
@@ -1,4 +1,22 @@
 @testset "Utils" begin
-  # TODO: readcoda
-  # TODO: compose
+  @testset "readcoda" begin
+    df = readcoda(joinpath(datadir,"juraset.csv"); codanames=[:Cd, :Cu, :Pb, :Co, :Cr, :Ni, :Zn]);
+    @test names(df) == ["X", "Y", "Rock", "Land", "Cd|Cu|Pb|Co|Cr|Ni|Zn"]
+    @test size(df) == (359, 5)
+
+    df = readcoda(joinpath(datadir,"juraset.csv"); codanames=["Cd", "Cu", "Pb", "Co", "Cr", "Ni", "Zn"]);
+    @test names(df) == ["X", "Y", "Rock", "Land", "Cd|Cu|Pb|Co|Cr|Ni|Zn"]
+    @test size(df) == (359, 5)
```

Comment:
Perhaps add a couple more tests for individual compositions. For example, make sure that the first 3 compositions are the expected ones from the file.